### PR TITLE
tests: performance and time tuning for e2e

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

- make check on CRD run in parallel
- add loop for check DSC status, so test on DSC status wont fail with timout
- move resourceCreationgTimeout and resourceRetryInterval out of tc
- rename resourceRetryInterval to generalRetryInterval
- create different timeout instead of using resourceCreationgTimeout for all cases
- move check on DSC ready out of testDSCCreation into validateDSCReady which calls waitDSCReady
- rename testAllApplicationCreation to testAllComponentCreation
- rename  testApplicationCreation to testComponentCreation
- do not immediate check func in testComponentCreation
- rename testApplicationDeletion to testComponentDeletion
- immediate check func in testComponentletion
- rename waitForControllerDeployment to waitForOperatorDeployment
- set new timeout, see const block

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by ykaliuta — APPROVED** [review]

File: tests/e2e/helper_test.go [hunk 1]
```diff
@@ -38,15 +39,24 @@ import (
...
+	deleteConfigMap          = "delete-configmap-name"
+	operatorReadyTimeout     = 2 * time.Minute
+	componentReadyTimeout    = 4 * time.Minute // currently in component wait is set from 2-3 mins, use 4 mins to not fail test.
+	componentDeletionTimeout = 1 * time.Minute
+	crdReadyTimeout          = 1 * time.Minute
+	csvWaitTimeout           = 1 * time.Minute
+	dsciCreationTimeout      = 20 * time.Second // time required to get a DSCI in Ready state
+	dscCreationTimeout       = 20 * time.Second // time required to wait till DSC in Ready state
+	generalRetryInterval     = 10 * time.Second
+	generalWaitTimeout       = 2 * time.Minute
```
Comment (ykaliuta) [inline tests/e2e/helper_test.go:55]: if this timeout is used for DSC Ready, it's too small from what I get. I'm struggling with existing 5 minutes now and it looks like it is too close to the limit.

Comment (zdtsw) [reply tests/e2e/helper_test.go:55]: `componentReadyTimeout    = 4 * time.Minute` is the one for the component to get Ready, and i put back the sleep time in the test before starting loop on component. 
XXXCreationTimeout is only for two CRs if they are created, not on component deployments

**Comment by zdtsw** [comment]
finally passed e2e with 
`ok  	github.com/opendatahub-io/opendatahub-operator/v2/tests/e2e	622.936s`

**Review by ykaliuta — APPROVED** [review]

**Pull request merged**